lib/ukio/apartments/handlers/availability_checker.ex

```diff
@@ -0,0 +1,20 @@
+defmodule Ukio.Apartments.Handlers.AvailabilityChecker do
+  import Ecto.Query, warn: false
+
+  alias Ukio.Apartments.Booking
+  alias Ukio.Repo
+
+  def is_apartment_available?(apartment_id, check_in, check_out) do
+    check_availability_query(apartment_id, check_in, check_out) |> Repo.all() |> Enum.empty?()
+  end
+
+  def check_availability_query(apartment_id, check_in, check_out) do
```

Comment:
This query will return any bookings that will overlap with the desired new booking dates.